CS9282: This member is not allowed in an extension block

**Version Used**:

.NET 10 RC 1

**Steps to Reproduce**:

Attempt to compile the extension indexer example from the documentation:

https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-14#extension-members

```cs
public static class Enumerable
{
    extension<TSource>(IEnumerable<TSource> source)
    {
        public bool IsEmpty => !source.Any(); // This is OK
        public TSource this[int index] => source.Skip(index).First(); // CS9282
        // public IEnumerable<TSource> Where(Func<TSource, bool> predicate) { ... }
    }
}
```

**Diagnostic Id**:

error CS9282: This member is not allowed in an extension block

**Expected Behavior**:

Being able to write extension indexers

**Actual Behavior**:

Documented extension indexer example does not build

--- a/docs/csharp/whats-new/csharp-14.md
+++ b/docs/csharp/whats-new/csharp-14.md
@@ -40,8 +40,6 @@ public static class Enumerable
     {
         // Extension property:
         public bool IsEmpty => !source.Any();
-        // Extension indexer:
-        public TSource this[int index] => source.Skip(index).First();
 
         // Extension method:
         public IEnumerable<TSource> Where(Func<TSource, bool> predicate) { ... }

